### ISSUE TITLE
p-calendar month picker displays the previously selected year, not the current one

### ISSUE THREAD

**Issue by szandra-homonnai** [primefaces/primeng#11038]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => Search github for a similar issue or PR before submitting
[ ] feature request => Please check if request is not on the roadmap already https://github.com/primefaces/primeng/wiki/Roadmap
[ ] support request => Please do not submit support request here, instead see http://forum.primefaces.org/viewforum.php?f=35
```

**Plunkr Case (Bug Reports)**
The original examples by PrimeNg: https://stackblitz.com/edit/primeng-calendar-demo

**Current behavior**
In the current p-calendar version, when selecting the year in the picker component, it goes automatically to the month selection, but in the header of month picker the year is wrong, it is not the newly selected year, but the previous one. After the month is selected and picker is closed, the good value is passed to the calendar, so it seems, that the emitted value is good, this error only effects the month picker part.

**Expected behavior**
After selecting a year, that should be in the header of the month picker.

**Minimal reproduction of the problem with instructions**
STEPS TO REPRODUCE:
In the inline example click on 2022, choose 2020 in the year picker, it automatically changes to month picker, but in the header it says 2022 as selected year not 2020.

**What is the motivation / use case for changing the behavior?**
This is misleading behaviour for users.

* **Angular version:** 13.0.3

* **PrimeNG version:** 13.0.4

* **Browser:** all

**Comment by aki-kesulahti** [comment]
Another steps to reproduce:
In the "Month Picker" example, click back or forward arrows in order to change year.
Currently the component header doesn't show the year change.

**Comment by yigitfindikli** [comment]
Fixed in 13.0.5.
